src/localization_protos.proto

```diff
@@ -2,12 +2,25 @@ syntax = "proto2";
 
 package image_sequence_localizer;
 
+// Assumes that all values of the matrix is known.
+// values.size() == rows*cols. No checks for this though.
 message CostMatrix {
   optional int32 rows = 20;
   optional int32 cols = 21;
   repeated double values = 1;
 }
 
+// Is used to store matching costs with associated query and reference id.
+// Does not require all elements of the cost matrix to be present
```

Comment:
These comments usually go out of date quite quickly and then become confusing to read. Maybe jut add the check whenever you read the message and remove this comment?